File: src-tauri/src/domain/completedworkout.rs
pub struct CompletedWorkout {
    pub workout_name: String,
    pub session_uuid: String,
    pub start_date: String,
    pub end_date: String,
}

pub type CompletedWorkouts = Vec<CompletedWorkout>;

// ######## For history view ######### //

use crate::domain::{SessionExercise, Set};
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone)]
pub struct DetailedWorkout {
    pub session_uuid: String,
    pub workout_uuid: String,
    pub workout_name: String,
    pub start_time: String,
    pub end_time: String,
    pub exercises: Vec<CompletedExercise>,
}

#[derive(Serialize, Deserialize, Clone)]
pub struct CompletedExercise {
    pub exercise_id: String,
    pub completed_id: String,
    pub name: String,
    pub body_part: String,
    pub gif_url: String,
    pub sets: Vec<CompletedSet>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(tag = "type")]
pub enum CompletedSet {
    Weighted { reps: f64, weight: f64 },
    Timed { distance: f64, time: f64 },
}

File: src/core/entities/CompletedWorkout.ts
import { DbDate } from "../../classes/dbDate";
import { ISessionExercises } from "../../types/types";

export default class CompletedWorkout {
      readonly workout_name: string;
      readonly session_uuid: string;
      readonly start_date: DbDate;
      readonly end_date: DbDate;
      readonly exercises?: ISessionExercises[];
    
    
    constructor(
        _workoutName: string,
        _sessionUuid: string,
        _startDate: string,
        _endDate: string,
        _exercises?: ISessionExercises[]
    )
    {
        this.workout_name = _workoutName;
        this.session_uuid = _sessionUuid;
        this.start_date = new DbDate(_startDate);
        this.end_date = new DbDate(_endDate);
        this.exercises = _exercises;
    }

    static fromDto() {
        
    }

}